File: cartracker/src/main/java/ch/rasc/cartracker/entity/option/Color.java
package ch.rasc.cartracker.entity.option;

import javax.persistence.Entity;

import ch.rasc.extclassgenerator.Model;

@Entity
@Model(value = "CarTracker.model.option.Color", readMethod = "colorOptionService.read",
		createMethod = "colorOptionService.create",
		updateMethod = "colorOptionService.update",
		destroyMethod = "colorOptionService.destroy", paging = true)
public class Color extends BaseOption {
	// nothing special
}

File: cartracker/src/main/webapp/app/model/option/Color.js
Ext.define("CarTracker.model.option.Color",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "longName",
    type : "string"
  }, {
    name : "shortName",
    type : "string"
  }, {
    name : "createDate",
    type : "date",
    dateFormat : "c"
  }, {
    name : "active",
    type : "boolean",
    defaultValue : true
  }, {
    name : "id",
    type : "int",
    useNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "colorOptionService.read",
      create : "colorOptionService.create",
      update : "colorOptionService.update",
      destroy : "colorOptionService.destroy"
    },
    reader : {
      root : "records"
    }
  }
});